File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/injection/module/AppModule.java
package com.benoitquenaudon.tvfoot.red.app.injection.module;

import android.app.Application;
import dagger.Module;
import dagger.Provides;
import io.reactivex.disposables.CompositeDisposable;

@Module public class AppModule {
  protected final Application application;

  public AppModule(Application application) {
    this.application = application;
  }

  @Provides Application provideApplication() {
    return application;
  }

  @Provides CompositeDisposable provideCompositeDisposable() {
    return new CompositeDisposable();
  }
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/injection/module/AppModule.kt
package com.benoitquenaudon.tvfoot.red.injection.module

import com.benoitquenaudon.tvfoot.red.injection.qualifiers.NowAtStartup
import dagger.Module
import dagger.Provides
import io.reactivex.disposables.CompositeDisposable
import java.util.concurrent.TimeUnit

@Module
object AppModule {
  @JvmStatic
  @Provides
  @NowAtStartup
  fun nowAtStartup(): Long = System.currentTimeMillis() - TimeUnit.HOURS.toMillis(3)

  @JvmStatic
  @Provides
  fun provideCompositeDisposable(): CompositeDisposable {
    return CompositeDisposable()
  }
}
